staging/src/k8s.io/kubelet/pkg/apis/dra/v1alpha3/api.proto

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// To regenerate api.pb.go run `hack/update-codegen.sh protobindings`
+
+syntax = "proto3";
+
+package v1alpha3;
+option go_package = "k8s.io/kubelet/pkg/apis/dra/v1alpha3";
+
+import "github.com/gogo/protobuf/gogoproto/gogo.proto";
+
+option (gogoproto.goproto_stringer_all) = false;
+option (gogoproto.stringer_all) =  true;
+option (gogoproto.goproto_getters_all) = true;
+option (gogoproto.marshaler_all) = true;
+option (gogoproto.sizer_all) = true;
+option (gogoproto.unmarshaler_all) = true;
+option (gogoproto.goproto_unrecognized_all) = false;
+
+service Node {
+  // NodePrepareResources prepares several ResourceClaims
+  // for use on the node. If an error is returned, the
+  // response is ignored. Failures for individidual claims
+  // can be reported inside NodePrepareResourcesResponse.
+  rpc NodePrepareResources (NodePrepareResourcesRequest)
+    returns (NodePrepareResourcesResponse) {}
+
+  // NodeUnprepareResources is the opposite of NodePrepareResources.
+  // The same error handling rules apply,
+  rpc NodeUnprepareResources (NodeUnprepareResourcesRequest)
+    returns (NodeUnprepareResourcesResponse) {}
+}
+
+message NodePrepareResourcesRequest {
+     // The list of ResourceClaims that are to be prepared.
+     repeated Claim claims = 1;
+}
+
+message NodePrepareResourcesResponse {
+    // The ResourceClaims for which preparation was done
+    // or attempted, with claim_uid as key.
+    //
+    // It is valid to return less entries. NodePrepareResources
```

Comment:
Is there a default number of tries before backing off if no response is returned for claim?

Is no response meant to indicate transient error, while explicit error should indicate definite error that cannot be overcome by a subsequent repetition of prepare call for that claim ?

Reply:
There's no such semantic difference. kubelet treats a missing response as if the prepare call hadn't been invoked and will retry like it normally does - at least that was the goal. I need to double-check whether the implementation really does it.

The expectation was that drivers only return less entries when there was some kind of error that they reported. I suspect that returning no error and less entries is not currently caught in the post-processing.

Reply:
The wording was indeed unfortunate. The intend was to treat that situation like an error, so the description should also state it like one. Changed.

The necessary check in the code was indeed missing - added.

While doing so, I found some cut-and-paste errors in error messages - fixed.

Here's the full diff: https://github.com/kubernetes/kubernetes/compare/3b74a711a02abc495b351499033593f9831959c3..fd74b7b8ddf7313af813f5c5e8cbb4923f69c534